prep_functions.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import pandas as pd
+from pathlib import Path
+from lxml import etree
+import cchardet as chardet
+
+
+class XML_Validator:
+
+    def __init__(self, xsd_path: str):
+        xmlschema_doc = etree.parse(
+            xsd_path)
+        self.xmlschema = etree.XMLSchema(
+            xmlschema_doc)
+
+    def validate(
+            self, xml_path: str) -> bool:
+        """_summary_ purpose checking for validated issues,
+        post download cycle on London Datastore,
+
+        :param xml_path: path to an xml file.
+        :type xml_path: str
+        :return:
+        :rtype: bool
+        """
+        xml_doc = etree.parse(xml_path)
+        result = self.xmlschema.validate(
+            xml_doc)
+
+        return result
+
+
+def file_encoding_audit(
+        data_folder: Path, ) -> pd.DataFrame:
+    """_summary_
+
+    :param data_folder: _description_
+    :type data_folder: Path
+    :return: _description_
+    :rtype: pd.DataFrame
+    """
+    # TODO - check csv encoding type of file, save to utf-8
+    # TODO - check xml encoding type of file,
+    # save as an acceptable format,
+    result_out = []
+    #
+    for cdf in data_folder.glob("**/*"):
+        if cdf.is_file() and 'log' not in cdf.root:
+            with open(cdf, "rb") as f:
+                msg = f.read()
+                result = chardet.detect(msg)
+                out = (
+                    f"{cdf.parts[-3]}, {cdf.stem}, {result}")
+                # this is messy.
+                outt = out.replace(
+                    '}', '').replace('{', '').replace('confidence', '').replace('encoding', '').replace("'':", "")
+                result_out.append(outt)
+    #
```

Comment:
save the outputs of the list generated by running cchardet on the file list, the result is then appended into a dataframe and filtered too return a list of files that have low confidence as to theire encoding types.